### ISSUE TITLE
Add "turnOffAfter" to switch type

### ISSUE THREAD

**Issue by alexisteleco** [ebaauw/homebridge-rpi#18]
Nice piece of software @ebaauw, thanks!

This is not an issue, but a petition

It would be great to have also a switch that turns of after x ms, so we can use it as if the raspberrypi was pressing a button and not a switch.
The button should do something like:
1. Turn ON
2. Wait certain period of time (100ms?)
3. Turn OFF

It would be an option like `turnOffAfter` in [mqttthing](https://github.com/arachnetech/homebridge-mqttthing#switch)

This way, it can be used to cicle between states, for example like old radios, with the search forward button.
It also would be useful for modernizing pieces of hardware with the typical three buttons "enter, next option, exit"
I would use it, for example, in modernizing old Hi-Fi systems, by bridging the standy button and connecting it to a raspberrypi relay; this way, I would be able to switch ON and OFF the Hi-Fi system as if I was pressing the physical button

What are your thougts about it?
Do you think it could be useful?

Again, thanks for the code :)

PS: I know this could be done with automation in "Home" app, with shortcuts on the iPhone, with python scripts... but each of them have its caveats and all of them seems a little bit "hacky"

**Comment by ebaauw** [comment]
Just so I understand correctly: you want to send a pulse to a device connected to an output GPIO pin?

**Comment by alexisteleco** [comment]
Thanks for your quick reply.

Yes, that would be completely the point

**Comment by ebaauw** [comment]
In v1.0.3.  Pulse is hard-coded to 20ms.

**Comment by alexisteleco** [comment]
Superb!
I installed Homebridge beta in order to try it, currently, I can't see the settings when I access through the Homebridge-UI and the "add block" button is doing nothing (without changing any setting).
<img width="789" alt="image" src="https://user-images.githubusercontent.com/17027081/80285001-6157c000-8722-11ea-9662-04e085c12a62.png">

Apart from this, when I add the pulse attribute in the `config.json` like this,
```json
{
    "platform": "RPi",
    "name": "RPi",
    "hosts": [
        {
            "host": "localhost:8888",
            "name": "Minicadena",
            "devices": [
                {
                    "device": "switch",
                    "name": "ON-OFF",
                    "pulse": true,
                    "gpio": 2
                }
            ]
        }
    ]
}
```
**It works as expected.**
Great work and thanks ;)

Nevertheless, I have to say that the hardcoded value is not suited for my personal case. I would need aprox 1 second.
¿Would you extend the feature in the future so the pulse width can be chosen?

PD: I can see you are the only contributor to this repo, and there are no "contributing" section in the readme, but in case a contribution is needed, or testing, or whatever, I would be glad to help.

Thanks again for quick responses and developing. Your work is appreciated.

**Comment by alexisteleco** [comment]
In the meantime, I changed the hardcoded value 20 to 1000 in my `/usr/local/lib/node_modules/homebridge-rpi/lib/RpiService.js` copy, reseted Homebridge and it working like a charm :)

**Comment by ebaauw** [comment]
>  I can't see the settings when I access through the Homebridge-UI

Oops.  This contributor shouldn’t edit json files without checking them...

> Nevertheless, I have to say that the hardcoded value is not suited for my personal case.

Happy to make it configurable In config.json, but when it’s longer than ~500ms the user might have time to flip back the switch.  I would want to limit it to 1000ms max, which is ridiculously long for a pulse.

**Comment by ebaauw** [comment]
In v1.0.4.

**Comment by alexisteleco** [comment]
> I would want to limit it to 1000ms max, which is ridiculously long for a pulse.

Completely agree; mine is a very particular case (also, I tried and ~500ms will also do the trick. Old stuff sometimes is not as bad as expected).

Thanks again for the quick reply and the even quicker change in the repo.

I will give this 1.0.4 a try tomorrow, and come back here to comment the results and close the issue.

**Comment by alexisteleco** [comment]
I tried 1.0.4 version. **It is working as expected**. 
The pulse setting shows a range selector which if you try to slide on *desktop* browser, you move the whole panel (tried on Safari and Chrome, MacOS, and Safari on iOS). I guess this is not a problem from this plugin.

<img width="680" alt="image" src="https://user-images.githubusercontent.com/17027081/80304249-fd320c00-87b4-11ea-9143-a5374ff3e605.png">

Just one thing to note, although it is not a problem itself, and should never happen if the property 'pulse' is well used:
If you try to press the switch twice faster than the selected pulse width, it remains in the ON state till you press it twice again.
In other words: **If you press twice quicker than pulse width, you reverse the switch logic.**

I am closing this issue as my petition is completely fulfilled.

Thanks!

**Comment by ebaauw** [comment]
So, as I said before, you want the pulse to be shorter than the time it takes the user to flip the switch off again.
